File: src/main/java/com/ddd/trainings/proposal/domain/Reviewer.java
package com.ddd.trainings.proposal.domain;

import com.ddd.utils.BaseEntity;
import lombok.Getter;

public class Reviewer extends BaseEntity {

  @Getter
  private final ReviewerId id;

  public Reviewer(ReviewerId id) {
    this.id = id;
  }
}

File: src/Ddd/Trainings/Proposal/Domain/Reviewer.cs
using Ddd.Utils;

namespace Ddd.Trainings.Proposal.Domain;

public class Reviewer : BaseEntity<ReviewerId>
{
    public ReviewerId Id { get; }
    public string Name { get; }
    public string[] Keywords { get; }
    public int CurrentReviewsCount { get; private set; }

    public Reviewer(ReviewerId id, string name, string[] keywords)
    {
        Id = id;
        Name = name;
        Keywords = keywords;
        CurrentReviewsCount = 0;
    }

    public override ReviewerId GetId()
    {
        return Id;
    }

    public void IncrementReviewsCount()
    {
        CurrentReviewsCount++;
    }
} 